drivers/wifi/infineon/airoc_wifi.c

```diff
@@ -0,0 +1,790 @@
+/*
+ * Copyright (c) 2023 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief AIROC Wi-Fi driver.
+ */
+
+#define DT_DRV_COMPAT infineon_airoc_wifi
+
+#include <zephyr/logging/log.h>
+#include <airoc_wifi.h>
+
+LOG_MODULE_REGISTER(infineon_airoc_wifi, CONFIG_WIFI_LOG_LEVEL);
+
+#ifndef AIROC_WIFI_TX_PACKET_POOL_COUNT
+#define AIROC_WIFI_TX_PACKET_POOL_COUNT (10)
+#endif
+
+#ifndef AIROC_WIFI_RX_PACKET_POOL_COUNT
+#define AIROC_WIFI_RX_PACKET_POOL_COUNT (10)
+#endif
+
+#ifndef AIROC_WIFI_PACKET_POOL_SIZE
+#define AIROC_WIFI_PACKET_POOL_SIZE     (1600)
+#endif
+
+#define AIROC_WIFI_PACKET_POOL_COUNT                                                               \
+	(AIROC_WIFI_TX_PACKET_POOL_COUNT + AIROC_WIFI_RX_PACKET_POOL_COUNT)
+
+/* This macro is copy of NET_BUF_POOL_FIXED_DEFINE with aligning net_buf_data_##_name
+ * WHD requires that network buffers is aligned, NET_BUF_POOL_FIXED_DEFINE does not
+ * guarantees aligned.
+ */
+#define NET_BUF_POOL_FIXED_DEFINE_ALIGN(_name, _count, _data_size, _ud_size, _destroy)             \
+	_NET_BUF_ARRAY_DEFINE(_name, _count, _ud_size);                                            \
+	static uint8_t __noinit net_buf_data_##_name[_count][_data_size] __net_buf_align;          \
+	static const struct net_buf_pool_fixed net_buf_fixed_##_name = {                           \
+		.data_size = _data_size,                                                           \
+		.data_pool = (uint8_t *)net_buf_data_##_name,                                      \
+	};                                                                                         \
+	static const struct net_buf_data_alloc net_buf_fixed_alloc_##_name = {                     \
+		.cb = &net_buf_fixed_cb,                                                           \
+		.alloc_data = (void *)&net_buf_fixed_##_name,                                      \
+	};                                                                                         \
+	static STRUCT_SECTION_ITERABLE(net_buf_pool, _name) = NET_BUF_POOL_INITIALIZER(            \
+		_name, &net_buf_fixed_alloc_##_name, _net_buf_##_name, _count, _ud_size, _destroy)
```

Comment:
The current main has the alignment supported in this macro so this can be removed.

Reply:
updated to use NET_BUF_POOL_FIXED_DEFINE.